views/js/qtiCreator/plugins/menu/preview.js

```diff
@@ -35,21 +35,38 @@ define([
 ], function(module, $, __, pluginFactory, hider, previewerFactory, buttonTpl){
     'use strict';
 
+    /**
+     * We assume the ClientLibConfigRegistry is filled up with something like this:
+     * 'taoQtiItem/qtiCreator/plugins/menu/preview' => [
+     *     'provider' => 'qtiItem',
+     * ],
+     *
+     * Or, with something like this for allowing multiple buttons in case of several providers are available:
+     * 'taoQtiItem/qtiCreator/plugins/menu/preview' => [
+     *     'provider' => 'qtiItem',
+     *     'providers' => [
+     *         ['id' => 'qtiItem', 'label' => 'Preview'],
+     *         ['id' => 'xxxx', 'label' => 'xxxx'],
+     *         ...
+     *     ],
+     * ],
```

Comment:
IMO we should replace it with a private method (`getProviders`) containing proper annotations documenting the contract. It will even make easier the development with autocomplete in the IDE.

Reply:
Yes, we could refactor the code and make it DRYier. However, it would go beyond the needs here. This is a "temporary" change, as it will support the transition to the new version of TAO.

Regarding the documentation, well, it as a free improvement. We don't have a central place where to manage this particular feature. It would be difficult to make it as you suggest, unfortunately.